starknet-rpc-test/get_transaction_by_hash.rs

```diff
@@ -0,0 +1,51 @@
+extern crate starknet_rpc_test;
+
+use rstest::rstest;
+use starknet_ff::FieldElement;
+use starknet_providers::Provider;
+use starknet_rpc_test::constants::{ARGENT_CONTRACT_ADDRESS, SIGNER_PRIVATE};
+use starknet_rpc_test::fixtures::madara;
+use starknet_rpc_test::utils::{create_account, AccountActions};
+use starknet_rpc_test::{MadaraClient, Transaction, TransactionResult};
+
+#[rstest]
+#[tokio::test]
+async fn work_valid_transaction_hash(#[future] madara: MadaraClient) -> Result<(), anyhow::Error> {
+    let madara = madara.await;
+    let rpc = madara.get_starknet_client();
+
+    let account = create_account(rpc, SIGNER_PRIVATE, ARGENT_CONTRACT_ADDRESS, true);
+    let txs = madara
+        .create_block_with_txs(vec![Transaction::Execution(account.transfer_tokens(
+            FieldElement::from_hex_be("0x123").unwrap(),
+            FieldElement::ONE,
+            None,
+        ))])
+        .await;
+
+    let rpc_response = match txs.unwrap().remove(0).unwrap() {
+        TransactionResult::Execution(rpc_response) => rpc_response,
+        _ => panic!("expected execution result"),
+    };
+
+    // there is a delay between the transaction being available at the client
+    // and the sealing of the block, hence sleeping for 100ms
+    tokio::time::sleep(tokio::time::Duration::from_millis(100)).await;
```

Comment:
That's quite unexpected behavior, I'm not sure I understand why this is happening

Reply:
So what I found from debugging is that `block_hash_from_transaction_hash` inside `mapping_db` returns None if you call it instantly after adding a transaction. And the db only writes the hashes when `sync_block` is called, which is called as a part of `mapping_sync_worker`. So, while `engine_createBlock` returns 200 it takes a few milli seconds for the blocks to be synced on the client db which is why we need to wait.

Reply:
Interesting, we didn't have this behavior within typescript tests. I suggest we move this to the create_block util then as it's quite related

Reply:
The delay is actually very small. Like, even if I add a print statement instead of a sleep, it still works. However, since this time would be machine dependent I added around 100ms.

I was thinking of adding it inside create_block but I think that will slow all other tests as well which is why I added it here. wdyt?

Reply:
Hmm, makes sense, will check how to do this

Reply:
Maybe a less platform-dependent thing would be to retry every 100ms, for 2 sec. And fail the test if after two sec it still returns an err.

Reply:
Ya I couldn't find any way to wait for the syncing either. @EvolveArt if you also agree, I can implement what @tdelabro mentioned

Reply:
@tdelabro, I discussed with @EvolveArt and have implemented the polling solution